File: Data Structures & Algorithms/trapping-rain-water/submission-1.js
class Solution {
    /**
     * @param {number[]} height
     * @return {number}
     */
    trap(arr) {
        let n = arr.length 
        let mxl = []
        let mxr = new Array(n)
    
        mxl.push(arr[0])
        for(let i = 1;i < n ;i++) mxl.push(Math.max(mxl[i-1],arr[i]))
        mxr[n-1] = arr[n-1]
        for(let i = n-2;i >=0 ;i--) mxr[i] = (Math.max(mxr[i+1],arr[i]))

        let ans = 0
        for(let i = 0 ;i < n ;i++){
            ans += Math.min(mxl[i],mxr[i]) - arr[i]
        }
        return ans 


    }
}

File: Data Structures & Algorithms/counting-bits/submission-1.java
class Solution {
    public int[] countBits(int n) {
        int ans[] = new int[n+1];
        for(int i = 0 ;i < n +1;i++){
            ans[i] = count(i);
        }
        return ans ;
    }
    private int count(int n){
        int a = 0;
        while(n!=0){
            n = n & (n-1);
            a++;
        }
        return a;
    }
}
